lib/middleware.js

```diff
@@ -12,28 +12,33 @@ const _hasParams = target => (...params) => (req, _, next) => {
 
 // for POST /cenreport
 // expect body-parser for JSON is registered
-module.exports.postCENReportReq = _hasParams('body')('report', 'cenKeys')
-module.exports.postCENReport = (req, _, next) => {
-  // expect postCENReportReq middleware is used before this one
+const postReportReq = _hasParams('body')('report')
+const postReport = (req, _, next) => {
+  // expect postReportReq middleware is used before this one
   const { report, cenKeys } = req.body
 
   let reportId
   try {
     reportId = computeHash(report)
```

Comment:
Question-1: if request body will be something like this:
 `{
  "validationPin": "string_token",
  "traces": [
   {
    "geohash": "dpz83d",
    "start": 1586791674,
    "end": 1586791950
   }
],
  "report": "c432bf9538f248a8238543a89d08e098c89bb890",
}`
where `report` comes in already encoded, why does it need to be hashed again here?

Question-2: why do we need a hashed `reportId`? Do both `report` and `reportId` need to be stored? if so why?

Reply:
`reportId` is probably not needed anymore with v4 and moving forward, as the trend indicates less and less fields and control exposed to server-side.